proposals/004-terminology-for-authentication.md

```diff
@@ -0,0 +1,115 @@
+# Terminology for Authentication
+
+This proposal defines terminology for describing authentication in the proxy. 
+It doesn't propose any changes to any code. 
+It is the first of a number of proposals which will use this terminology.
+
+For the purposes of this proposal we consider TLS and SASL as the only authentication mechanisms we care about.
+In the following, the word "component" is generalising over filters, other plugins, and a proxy virtual cluster as a whole.
+
+## Definitions
+
+### Mutual authentication
+
+A **mutual authentication mechanism** is one which proves the identity of a server to a client.
```

Comment:
Is it missing a bit like, and proves the identity of a client to a server? Think theres a 'mutual' aspect missing.